notify/impl.go

```diff
@@ -390,24 +399,31 @@ func (n *PagerDuty) Notify(ctx context.Context, as ...*types.Alert) error {
 		msg.ClientURL = tmpl(n.conf.ClientURL)
 	}
 	if err != nil {
-		return err
+		return false, err
 	}
 
 	var buf bytes.Buffer
 	if err := json.NewEncoder(&buf).Encode(msg); err != nil {
-		return err
+		return false, err
 	}
 
 	resp, err := ctxhttp.Post(ctx, http.DefaultClient, n.conf.URL, contentTypeJSON, &buf)
 	if err != nil {
-		return err
+		return false, err
 	}
 	resp.Body.Close()
 
+	// retry can solve the issue on 403 (rate limiting) and 5xx response codes
+	// https://v2.developer.pagerduty.com/docs/trigger-events
+	if resp.StatusCode == 403 || resp.StatusCode/100 == 5 {
```

Comment:
Oh my god – 403 for rate limiting?
I was just thinking about generalizing the retry inspection in its own function...